arch/x86/boot/compressed/sl_stub.S

```diff
@@ -276,12 +276,14 @@ SYM_FUNC_START(sl_txt_verify_os_mle_struct)
 	cmpl	%esi, %edx
 	ja	.Lbuffer_beyond_pmr
 
-	/* Check the boot params */
+	/*
+	 * Check the boot params. Note during a UEFI boot, the boot
+	 * params will be inside the MLE image so there is no overlap check here.
+	 */
 	movl	SL_boot_params_addr(%eax), %ecx
 	movl	$(PAGE_SIZE), %edx
 	addl	%ecx, %edx
 	jc	.Loverflow_detected
-	call	sl_check_buffer_mle_overlap
```

Comment:
is this the only call, and thus it can be removed?

Reply:
No, also I may revisit this and enhance the logic a bit. I want to re-add the check and make it smarter.